libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -32,34 +32,39 @@ public static class ServiceCollectionExtensions
         /// </summary>
         /// <remark>
         /// The following dependencies are added with TrySingleton so advanced scenarios can override them to customize the runtimne behavior:
-        /// BotFrameworkAuthentication,
-        /// IBot,
-        /// ResourceExplorer,
-        /// LanguagePolicy,
-        /// ChannelServiceHandlerBase,
-        /// IStorage,
-        /// UserState,
-        /// ConversationState,
-        /// SkillConversationIdFactoryBase
-        /// and IBotTelemetryClient.
+        /// <see cref="BotFrameworkAuthentication"/>,
+        /// <see cref="IBot"/>,
+        /// <see cref="ResourceExplorer"/>,
+        /// <see cref="LanguagePolicy"/>,
+        /// <see cref="ChannelServiceHandlerBase"/>,
+        /// <see cref="IStorage"/>,
+        /// <see cref="UserState"/>,
+        /// <see cref="ConversationState"/>,
+        /// <see cref="SkillConversationIdFactoryBase"/>
+        /// and <see cref="IBotTelemetryClient"/>.
+        /// 
+        /// While the full set of dependencies is designed to be sufficient to run Adaptiev Dialogs the <see cref="IBot"/> dependency can actually be
+        /// any <see cref="IBot"/> implementation and is not constrained to one that uses Adaptive Dialogs. Any Bot Framework project Startup can therefore
+        /// be simplified by just using this function along with a custom <see cref="IBot"/> implementation.
         /// 
         /// Aspects of the behavior of a number of these dependencies, including those that can be overriden, can be controlled through configuration.   
         /// 
-        /// The default ResourceExlorer uses the file system. The folder used being read from configuration.
+        /// The default <see cref="ResourceExplorer"/> uses the file system. The folder used being read from configuration.
         /// 
-        /// The default LanguagePolicy is "us-en" and this can be changed through configuration.
+        /// The default <see cref="LanguagePolicy"/> is "us-en" and this can be changed through configuration.
         /// 
-        /// If not overriden, the exact type of storage added depends on configuration. With no configuration the default is memory storage.
-        /// It should be noted that MemoryStorage is designed primarily for testing with a single host running the bot and no database.
+        /// If not overriden, the exact type of <see cref="IStorage"/> added depends on configuration. With no configuration the default is memory storage.
+        /// It should be noted that <see cref="MemoryStorage"/> is designed primarily for testing with a single host running the bot and no database.
```

Comment:
Nit: could replace 'database' with 'storage' or 'durable storage' perhaps? Just a thought, not a big deal.

Reply:
i was trying to avoid using the same work, I'll replace with durable storage - "persistence" being another word I guess